### PR TITLE
server: update subjects

### LINKED ISSUE #423: Delete old subjects
Please delete
- Musik
-BWR (Betriebswirtschaftslehre mit Rechnungswesen)
- Deutsch als Fremdsprache

### PR DESCRIPTION
**Pull request by inyono**
Not yet finished, need to test out some stuff locally.

Close #423 #516

### CODE REVIEW

File: packages/public/server/migrations/20201020090526-update-subjects.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            if (err) {
+              return cb(err)
+            }
+            createSubject(db, { instance: 1, name: 'Lerntipps' }, cb)
+          })
+        }
+      )
+    }
+  )
+}
```
Comment (inyono) [inline packages/public/server/migrations/20201020090526-update-subjects.js:53]: We might want to write some helpers for that / add a TS toolchain. Code gets quite verbose now.

**Comment by inyono** [comment]
If possible, we should also handle the navigation.

File: packages/public/server/migrations/20201020090526-update-subjects.js (hunk 1 above)
Comment (kulla) [reply packages/public/server/migrations/20201020090526-update-subjects.js:53]: Seems like a Monad-structure to me :smile:

File: packages/migrations/src/20200220155526-migrate-exercises.ts [hunk 2]
```diff
@@ -20,58 +20,48 @@
...
+      )
+      await processResults(remainingResults)
+    }
+
+    const results = await db.runSql<Row[]>(`
       SELECT erf.id, erf.value
       FROM entity_revision_field erf
-        LEFT JOIN entity_revision er on erf.entity_revision_id = er.id
-        LEFT JOIN entity e on er.repository_id = e.id
+             LEFT JOIN entity_revision er on erf.entity_revision_id = er.id
```
Comment (inyono) [inline packages/migrations/src/20200220155526-migrate-exercises.ts:52]: weird spacing.

File: packages/migrations/src/20201023104526-update-subjects.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    term,
+    subjectTaxonomy
+  )
+}
+
+function removeSubject(
+  db: Database,
+  { instance, name }: { instance: number; name: string }
+) {
+  // TODO: also remove navigation
```
Comment (inyono) [inline packages/migrations/src/20201023104526-update-subjects.ts:82]: @inyono

File: packages/migrations/src/20201023104526-update-subjects.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  // TODO: clear dead uuids (since this query will trigger some deletions) ;)
```
Comment (inyono) [inline packages/migrations/src/20201023104526-update-subjects.ts:83]: @inyono

**Pull request merged**